src/server/stream_family.cc

```diff
@@ -636,6 +689,52 @@ OpResult<RecordVec> OpRange(const OpArgs& op_args, string_view key, const RangeO
   return result;
 }
 
+OpResult<RecordVec> OpRangeFromConsumerPEL(const OpArgs& op_args, string_view key,
+                                           const RangeOpts& opts) {
+  RecordVec result;
+
+  if (opts.count == 0)
+    return result;
+
+  unsigned char start_key[sizeof(streamID)];
+  unsigned char end_key[sizeof(streamID)];
+  auto sstart = opts.start.val;
+  auto send = opts.end.val;
+
+  StreamEncodeID(start_key, &sstart);
+  StreamEncodeID(end_key, &send);
+  raxIterator ri;
+
+  raxStart(&ri, opts.consumer->pel);
+  raxSeek(&ri, ">=", start_key, sizeof(start_key));
+  size_t ecount = 0;
+  while (raxNext(&ri) && (!opts.count || ecount < opts.count)) {
+    if (memcmp(ri.key, &send, ri.key_len) > 0)
+      break;
+    streamID id;
+
+    streamDecodeID(ri.key, &id);
+    RangeOpts ropts;
+    ropts.start.val = id;
+    ropts.end.val = id;
+    auto op_result = OpRange(op_args, key, ropts);
+    if (!op_result) {
+      Record rec;
+      rec.id = id;
+      rec.kv_arr.clear();
+      result.push_back(move(rec));
+    } else {
+      streamNACK* nack = static_cast<streamNACK*>(ri.data);
+      nack->delivery_time = mstime();
+      nack->delivery_count++;
+      result.push_back(move(op_result.value()[0]));
```

Comment:
This is ok because in each iteration, `OpRange` gives a new `Recordvec` containing  a new `Record`. So on each iteration, it would move distinct `Record` to `result`.